fenwick/src/intTest/java/org/opencadc/fenwick/InventoryHarvesterTest.java

```diff
@@ -415,4 +414,62 @@ public void runIncludeArtifacts() throws Exception {
 
         Assert.assertNotNull("included", inventoryEnvironment.artifactDAO.get(artifactTwo.getID()));
     }
+    
+    @Test
+    public void atestCollisions() throws Exception {
```

Comment:
oh, yeah, I was trying to detect a case with an acidentally open txn so changed the method name so it would run earlier and cause something else to hang (which it did). On that note: I re-did the try/catch/finally around transactions so they better defended against accidentally jumping out of the txn without commit or rollback.  It will also make it easier to improve the `catch(Exception` block in future because it is fully inside the loop.

I'll push that commit in here now as well.